src/DataHydrator.php

```diff
@@ -27,7 +31,7 @@ class DataHydrator implements DataResolverInterface {
     public const KEY_ROOT = '$root';
 
     /**
-     * @var DataResolverInterface[]
+     * @var AbstractDataResolver[]
```

Comment:
Is this correct? Have you gotten rid of the interface altogether?

Reply:
Not quite. The data resolver interface is still used in intermediate steps (like in middlewares). It just can't be registered (registered ones are needed to build a schema).

So the current `DataHydrator::$resolver` will always be just the interface (because of the anonymous class defined in `DataHydrator::makeResolver()` implementing just the interface when composed with middlewares.

Registered ones need to provide additional information though, so the `addResolver()` and `getResolver()` only work with `AbstractDataResolver`.